### ISSUE TITLE
Ability to define a union between mongoId and a class (and possibly between other primitive types)

### ISSUE THREAD

**Issue by Rush** [deepkit/deepkit-framework#81]
Use case:
A field `author` can be either an `ObjectId` or a populated user.

```ts
class ResolvedAuthor {
  @t
  avatar?: string;
  @t
  email!: string;
  @t
  displayName?: string;
  @t
  verified!: boolean;
}

class Entity {
    @t.union(t.mongoId, ResolvedAuthor)
    author!: Types.ObjectId | ResolvedAuthor;
}
```

But this doesn't work as union requires a discriminant:
```
(node:32334) UnhandledPromiseRejectionWarning: Error: Type of property author_1 has no discriminant or literal set. Could not discriminate the value.
```

**Comment by marcj** [comment]
Yes, we could optimize the union type detection for simple unions like that.

Internal note: This should happen in https://github.com/deepkit/deepkit-framework/blob/master/packages/type/src/union.ts#L47, by introducing a new simple `object` checker. When the union properties contain only one 'class' type, we switch to that object checker for this particular type, which then basically bypasses the more concrete discriminator checks.

**Comment by marcj** [comment]
Btw, `@t.mongoId` returns a `string, not `ObjectId`, but that just as a side note.

**Comment by marcj** [comment]
Fixed in https://github.com/deepkit/deepkit-framework/commit/baf1b5394d05f96a18082445eea0349b8bc84bf6

**Comment by marcj** [comment]
released in 1.0.1-alpha.4
